src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -744,7 +751,19 @@ private bool TakeStartLine(SocketInput input)
                 RequestUri = requestUrlPath;
                 QueryString = queryString;
                 HttpVersion = httpVersion;
-                Path = RequestUri;
+
+                if (!string.IsNullOrEmpty(_pathBase) &&
+                    (requestUrlPath.Equals(_pathBase, StringComparison.OrdinalIgnoreCase) ||
+                     requestUrlPath.StartsWith(_pathBaseWithSlash, StringComparison.OrdinalIgnoreCase)))
+                {
+                    PathBase = requestUrlPath.Substring(0, _pathBase.Length);
+                    Path = requestUrlPath.Substring(_pathBase.Length);
```

Comment:
I'm going to rewrite this to use iterators and also avoid scanning the string multiple times.